crates/sys/src/lib.rs

```diff
@@ -48,21 +48,79 @@ mod bindings;
 /// ```
 /// flipperzero_sys::crash!("Hello world!");
 /// ```
+///
+/// Crash the system with default *"Fatal Error"* message:
+///
+/// ```
+/// flipperzero_sys::crash!();
+/// ```
 #[macro_export]
 macro_rules! crash {
-    ($msg:expr $(,)?) => {{
-        const MESSAGE: *const ::core::primitive::i8 =
-            match ::core::ffi::CStr::from_bytes_with_nul(
-                ::core::concat!($msg, "\0").as_bytes(),
-            ) {
-                Ok(cstr) => cstr.as_ptr(),
-                Err(error) => panic!("message contains NULs"),
+    ($($msg:expr)? $(,)?) => {{
+        // Optional message
+        let mut message: *const i8 = ::core::ptr::null();
+        $(
+            message = match ::core::ffi::CStr::from_bytes_with_nul(::core::concat!($msg, "\0").as_bytes()) {
```

Comment:
This is out of  this PR's scope, but  it may be worth switching to `CStr` for `$msg`

Reply:
I was wondering about that too. Certainly much less painful now that there's `CStr` literals.

(Funny enough, it's really hard to see these messages without using the hardware UART. Usually the USB connection drops before the message gets written to the USB TTY).